source/AI.cpp

```diff
@@ -769,8 +769,11 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 			// A carried ship must belong to the same government as its parent to dock with it.
 			bool hasParent = parent && !parent->IsDestroyed() && parent->GetGovernment() == gov;
 			bool inParentSystem = hasParent && parent->GetSystem() == it->GetSystem();
-			bool parentHasSpace = inParentSystem && parent->BaysFree(it->Attributes().Category());
-			if(!hasParent || (!inParentSystem && !it->JumpFuel()) || (!parentHasSpace && !Random::Int(1800)))
+			// NPCs may take 30 seconds or longer to find a new parent.  Player
+			// owned fighter shouldn't take more than a few seconds.
+			bool findNewParent = it->IsYours() ? !Random::Int(30) : !Random::Int(1800);
+			bool parentHasSpace = !findNewParent || (inParentSystem && parent->CanCarry(*it));
```

Comment:
```suggestion
			bool parentHasSpace = (inParentSystem && parent->CanCarry(*it));
```

Not sure why you need that condition here. Also the name doesn't fit anymore.

Reply:
Because why waste the CPU cycles calculating when it will fail due to random int?  CanCarry iterates across all escorts and is an expensive operation.  By setting it to boolean true before bothering to calculate it you skip that expensive calculation.

It gets ignored one line later